client/galaxy/scripts/entry/panels/tool-panel.js

```diff
@@ -52,9 +52,22 @@ const ToolPanel = Backbone.View.extend({
                 title: _l("Show favorites"),
                 icon: "fa fa-star-o",
                 onclick: e => {
-                    $("#tool-search-query")
-                        .val("#favorites")
-                        .trigger("change");
+                    var search_query = $("#tool-search-query");
+                    var header_btn = $(".panel-header-button");
```

Comment:
You can use `const` instead of `var`. In general, please use either `let` or `const` to declare variables. 
ping @janani-prasad